### ISSUE TITLE
Resolve cyclical SQL join references in preset `cell-health-cellprofiler-to-cytominer-database`

### ISSUE THREAD

**Issue by d33bs** [cytomining/CytoTable#105]
The preset `cell-health-cellprofiler-to-cytominer-database` includes self-referential join SQL logic which should be resolved. For example, `cells.Metadata_TableNumber = cells.Metadata_TableNumber`.

Originally mentioned by @shntnu and @Zitong-Chen-16 (many thanks!).

**Comment by shntnu** [comment]
> Originally mentioned by @shntnu (many thanks!).

h/t @Zitong-Chen-16

**Comment by d33bs** [comment]
Just an update: I've sent along a fix for this in #106 . As part of the work for the PR I also sought to help automatically detect or prevent this sort of issue via SQL linting and special handling of SQL string types within CytoTable (#107). I checked SQLFluff for this capability today and didn't yet see it there so submitted https://github.com/sqlfluff/sqlfluff/issues/5213 . Even if we cannot make use of the functionality there today, it might be that we (or others) eventually could benefit from this being implemented as a static analysis check.

**Comment by Zitong-Chen-16** [comment]
Thank you so much for your help, @d33bs! I would like to share that I was able to use Cytotable on another dataset using the following [script](https://gist.github.com/Zitong-Chen-16/8a48675a784d372ec05ddd9b69a9d248) though not without changes to the parameters that were passed in. Ane downstream analysis confirmed that the files were merged successfully.

For my specific dataset, I had to change the `identifying_cols ` and `join_command ` in order for it to work with the `cell-health-cellprofiler-to-cytominer-database` preset. Specifically, I changed `identifying_cols ` to include `ObjectNumber ` and removed `Cells_ObjectNumber` and `Nuclei_ObjectNumber`, and was wondering if that would actually make more sense in a general setting since IIUC ObjectNumber is a shared metadata column across the tables. For the `join_command` I changed the `Image_Metadata_Well` and `Image_Metadata_Plate` [here](https://github.com/cytomining/CytoTable/blob/d64f954edce640d2559defc49be3a099b07bb4e5/cytotable/presets.py#L193C10-L193C10) to `Metadata_Well` and Metadata_Plate`, but not quite sure if that is dataset specific.

Additionally, I also encountered a package dependency issue. Specifically, the script was able to run when I uninstalled `pandas`, but gave an error when `pandas==2.1.1` is installed. I am not sure if others have encountered similar issues or if I did something wrong. Thank you!
